### PR TITLE
Clean up project files

### PR DESCRIPTION
**Pull request by manne**
**What this PR does / why we need it**:
To clean up the project files

**Which issue(s) this PR fixes**:

Please reference the issue this PR will close: #506

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
should not

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests.
  - Changes should not affect behavior
- [x] Unit tests passing
- [ ] End-to-end tests passing
  - Changes should not affect behavior
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by manne** [comment]
For the one who is interested, how the properties can easily be removed, can use this regular expression `^\s*<RepositoryUrl>\$\(RepositoryUrl\)<\/RepositoryUrl>\s*$\n` and replace it occurrence with an empty string.

**Comment by sungam3r** [comment]
PackageReference to `Microsoft.SourceLink.GitHub` can also be moved.

**Comment by manne** [comment]
I removed the `LangVersion` from `Directory.Build.props`, because it is already present in `build/dependencies.props`

**Comment by unaizorrilla** [comment]
Hi @manne 

Make sense, let me a couple of days to merge it, thanks for contribute!!

**Comment by manne** [comment]
@unaizorrilla you are welcome.

File: Directory.Build.props [hunk 1]
```diff
@@ -7,6 +7,13 @@
     <EnableSourceControlManagerQueries>false</EnableSourceControlManagerQueries>
   </PropertyGroup>
   
+  <ItemGroup>
```
Comment (unaizorrilla) [inline Directory.Build.props:10]: Why EnableSourceLink is disabled? we disabled only for non windows os but will be enabled by default

Comment (manne) [reply Directory.Build.props:10]: > Why EnableSourceLink is disabled? we disabled only for non windows os but will be enabled by default

Sorry. I do not understand your question and statement. Did I unintentionally changed the state of `EnableSourceLink`?

Comment (unaizorrilla) [reply Directory.Build.props:10]: I'm sorry sure I need to improve my english :-)

Well, the first lines on this file remove the source link, but after check the history the problem is not your. Can yoy remove the initial  property group?

  
  <PropertyGroup>
    <EnableSourceLink>false</EnableSourceLink>
    <EnableSourceControlManagerQueries>false</EnableSourceControlManagerQueries>
  </PropertyGroup>

Comment (manne) [reply Directory.Build.props:5]: @unaizorrilla is this change OK for you?

Comment (unaizorrilla) [reply Directory.Build.props:5]: looks good, I try to merge tomorrow!

**Comment by unaizorrilla** [comment]
Merged, thanks @manne for contribute and improve on this!

**Pull request merged**